### PR TITLE
Convert DeleteSynonymGroupEntityTest to MFTF

### PR DESCRIPTION
**Pull request by kate-kyzyma**
### Fixed Issues

1. magento/magento-functional-tests-migration#389: Convert DeleteSynonymGroupEntityTest to MFTF

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Search/Test/Mftf/Data/SynonymGroupData.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  */
+-->
+
+<entities xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="urn:magento:mftf:DataGenerator/etc/dataProfileSchema.xsd">
+    <entity name="search_group_1">
+        <data key="scope">0</data>
+        <data key="synonyms" unique="suffix">synonym_</data>
+    </entity>
+
```
Comment (lbajsarowicz) [inline app/code/Magento/Search/Test/Mftf/Data/SynonymGroupData.xml]: Just redundant space :-)

File: app/code/Magento/Search/Test/Mftf/Section/AdminNewSynonymsGroupFormSection.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<sections xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/SectionObject.xsd">
+    <section name="AdminNewSynonymsGroupFormSection">
+        <element name="scopeValue" type="select" selector="//select[@class='admin__control-select']//option[contains(., '{{var}}')]" parameterized="true" />
```
Comment (lbajsarowicz) [inline app/code/Magento/Search/Test/Mftf/Section/AdminNewSynonymsGroupFormSection.xml]: Could you replace `{{var}}` with something more self-descriptive?

File: app/code/Magento/Search/Test/Mftf/Section/AdminSearchSynonymsGridSection.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  * See COPYING.txt for license details.
+  */
+-->
+
+<sections xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/SectionObject.xsd">
+    <section name="AdminSearchSynonymsGridSection">
+        <element name="newSynonymGroup" type="button" selector="button[title='New Synonym Group']"/>
+        <element name="column" type="text" selector="//tr//td[count(//div[@data-role='grid-wrapper']//tr//th[contains(., '{{col}}')]/preceding-sibling::th) +1 ]" parameterized="true"/>
+        <element name="selectFirstRow" type="checkbox" selector="//td[@class='data-grid-checkbox-cell']"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Search/Test/Mftf/Section/AdminSearchSynonymsGridSection.xml]: Element should be noun, not verb :-)

File: app/code/Magento/Search/Test/Mftf/Section/AdminSearchSynonymsGridSection.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        <element name="actions" type="button" selector="//div[@class='col-xs-2']/div[@class='action-select-wrap']/button[@class='action-select']" timeout="30"/>
+        <element name="delete" type="button" selector="//*[contains(@class,'admin__data-grid-header-row row row-gutter')]//*[text()='Delete']" timeout="30"/>
+        <element name="ok" type="button" selector="//button[@data-role='action']//span[text()='OK']" timeout="30"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Search/Test/Mftf/Section/AdminSearchSynonymsGridSection.xml]: `OK` is button in confirmation modal? If so - more descriptive would be `modalConfirmationButton`.

File: app/code/Magento/Search/Test/Mftf/Test/AdminDeleteSynonymGroupEntityTest.xml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminDeleteSynonymGroupEntityTest">
+        <annotations>
+            <features value="Search"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Search/Test/Mftf/Test/AdminDeleteSynonymGroupEntityTest.xml]: Try with `Search Synonyms`

File: app/code/Magento/Search/Test/Mftf/Test/AdminDeleteSynonymGroupEntityTest.xml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            <stories value="Admin delete synonym group"/>
+            <title value="Admin delete synonym group"/>
+            <description value="Admin delete synonym group"/>
+            <group value="search"/>
+            <group value="mtf_migrated"/>
+        </annotations>
+
+        <before>
+            <actionGroup ref="LoginAsAdmin" stepKey="loginAsAdmin"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Search/Test/Mftf/Test/AdminDeleteSynonymGroupEntityTest.xml]: `AdminLoginActionGroup`

File: app/code/Magento/Search/Test/Mftf/Test/AdminDeleteSynonymGroupEntityTest.xml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        </before>
+        <after>
+            <actionGroup ref="logout" stepKey="logout"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Search/Test/Mftf/Test/AdminDeleteSynonymGroupEntityTest.xml]: `AdminLogoutActionGroup`

File: app/code/Magento/Search/Test/Mftf/Data/SynonymGroupData.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<entities xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="urn:magento:mftf:DataGenerator/etc/dataProfileSchema.xsd">
+    <entity name="search_group_1">
```
Comment (lbajsarowicz) [inline app/code/Magento/Search/Test/Mftf/Data/SynonymGroupData.xml]: We should be consistent with naming.
Most of the system already have `PascalCaseNaming` and I think we should follow that pattern.

**Comment by kate-kyzyma** [comment]
Hi @lbajsarowicz!
Thank you for reviewing my test.
I fixed all mentioned issues except the one about using  {{var}} value. I use it to make the selector more flexible as we can pass different scope value, for example, All websites or a proper store view. Do you want me to modify it to use only one value (All websites) suitable for this test?

Thanks.

**Comment by lbajsarowicz** [comment]
I just want you to replace "var" with something more meaningful like "scopeName" or "scopeKey", not removing the variable

**Comment by kate-kyzyma** [comment]
@lbajsarowicz 
Thank you for the explanation. I have fixed the last issue.

Thanks.

**Pull request merged**